### PR TITLE
Updates deprecated Alpaca RestClient to AlpacaTradeClient/PolygonDataClient

### PR DESCRIPTION
**Pull request by gsalaz98**
#### Description

* Updates interfaces and response types to ensure compatibility with new API
* Updates the `RestClient` to `AlpacaTradeClient` and `PolygonDataClient`
#### Related Issue

N/A
#### Motivation and Context

Live stability
#### Requires Documentation Change

No
#### How Has This Been Tested?

* Cloud testing (paper and live)
* local testing (paper) - tested orders and data feed
* local testing (live) - tested data feed
#### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Refactor (non-breaking change which improves implementation)
- [ ] Performance (non-breaking change which improves performance. Please add associated performance test and results)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Non-functional change (xml comments/documentation/etc)

#### Checklist:

- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** [document](https://github.com/QuantConnect/Lean/blob/master/CONTRIBUTING.md).
- [ ] I have added tests to cover my changes. 
- [x] All new and existing tests passed.
- [ ] My branch follows the naming convention `bug-<issue#>-<description>` or `feature-<issue#>-<description>`

### CODE REVIEW

File: Brokerages/Alpaca/Markets/PolygonDataClient.cs [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+                .EnsureIsValid();
+
+            _isStagingEnvironment = configuration.KeyId
+                .EndsWith("-staging", StringComparison.Ordinal);
+            _keyId = configuration.KeyId;
+
+            _httpClient.DefaultRequestHeaders.Accept
+                .Add(new MediaTypeWithQualityHeaderValue("application/json"));
+            _httpClient.BaseAddress = configuration.ApiEndpoint;
+            _httpClient.SetSecurityProtocol();
```
Comment (StefanoRaggi) [inline Brokerages/Alpaca/Markets/PolygonDataClient.cs]: Seems this line will not be compiled unless we `#define NET45` ?
https://github.com/QuantConnect/Lean/pull/4228/files#diff-d6fdf1446f3e7eab9248163398e45ec8R141

File: Brokerages/Alpaca/Markets/AlpacaTradingClient.cs [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+            _alpacaRestApiThrottler = configuration.ThrottleParameters.GetThrottler();
+
+            _httpClient.AddAuthenticationHeaders(configuration.SecurityId);
+
+            _httpClient.DefaultRequestHeaders.Accept
+                .Add(new MediaTypeWithQualityHeaderValue("application/json"));
+            _httpClient.BaseAddress = configuration.ApiEndpoint
+                .AddApiVersionNumberSafe(configuration.ApiVersion);
+            _httpClient.SetSecurityProtocol();
```
Comment (StefanoRaggi) [inline Brokerages/Alpaca/Markets/AlpacaTradingClient.cs]: Seems this line will not be compiled unless we `#define NET45` ?
https://github.com/QuantConnect/Lean/pull/4228/files#diff-d6fdf1446f3e7eab9248163398e45ec8R141

File: Brokerages/Alpaca/Markets/PolygonDataClient.cs (hunk 1 above)
Comment (gsalaz98) [reply Brokerages/Alpaca/Markets/PolygonDataClient.cs]: I define the NET45 constant at the top of the file on line 1

File: Brokerages/Alpaca/Markets/AlpacaTradingClient.cs (hunk 2 above)
Comment (gsalaz98) [reply Brokerages/Alpaca/Markets/AlpacaTradingClient.cs]: I define the NET45 constant at the top of the file on line 1

File: Brokerages/Alpaca/Markets/PolygonDataClient.cs (hunk 1 above)
Comment (StefanoRaggi) [reply Brokerages/Alpaca/Markets/PolygonDataClient.cs]: It's needed in `Brokerages/Alpaca/Markets/PolygonDataClient.cs`

File: Brokerages/Alpaca/Markets/AlpacaTradingClient.cs (hunk 2 above)
Comment (StefanoRaggi) [reply Brokerages/Alpaca/Markets/AlpacaTradingClient.cs]: It's needed in `Brokerages/Alpaca/Markets/AlpacaTradingClient.cs`

File: Brokerages/Alpaca/Markets/RestClient.Polygon.cs [hunk 3]
```diff
@@ -107,34 +107,35 @@ public Task<IEnumerable<IExchange>> ListExchangesAsync()
...
-                    .AddParameter("to", dateIntoInclusive, "yyyy-MM-dd")
-                    .AddParameter("limit", limit)
-            };
-
-            return getSingleObjectAsync
-                <IAggHistoricalItems<IAgg>,
-                    JsonAggHistoricalItems<IAgg, JsonDayAgg>>(
-                _polygonHttpClient, FakeThrottler.Instance, builder);
-        }
+        // Commented out because JsonDayAgg doesn't exist anymore
```
Comment (Martin-Molinero) [inline Brokerages/Alpaca/Markets/RestClient.Polygon.cs]: Does this probably mean this file should be also updated?

File: Brokerages/QuantConnect.Brokerages.csproj [hunk 4]
```diff
@@ -270,21 +301,38 @@
...
+    <Compile Include="Alpaca\Markets\Parameters\ListOrdersRequest.cs" />
+    <Compile Include="Alpaca\Markets\Parameters\NewOrderRequest.cs" />
+    <Compile Include="Alpaca\Markets\Parameters\PolygonDataClientConfiguration.cs" />
     <Compile Include="Alpaca\Markets\Parameters\PolygonStreamingClientConfiguration.cs" />
+    <Compile Include="Alpaca\Markets\Parameters\PortfolioHistoryRequest.cs" />
     <Compile Include="Alpaca\Markets\Parameters\StreamingClientConfiguration.cs" />
+    <Compile Include="Alpaca\Markets\PolygonDataClient.cs" />
     <Compile Include="Alpaca\Markets\PolygonStreamingClient.cs" />
     <Compile Include="Alpaca\Markets\RateThrottler.cs" />
     <Compile Include="Alpaca\Markets\RestClient.cs" />
```
Comment (Martin-Molinero) [inline Brokerages/QuantConnect.Brokerages.csproj]: I see a lot of new files added but only a couple removed, should we remove more, like for example `RestClient`?

**Pull request merged**